component/azstorage/file_share.go

```diff
@@ -150,86 +154,259 @@ func (fs *FileShare) TestPipeline() error {
 }
 
 func (fs *FileShare) ListContainers() ([]string, error) {
-	return nil, nil
+	log.Trace("FileShare::ListContainers : Listing containers")
+	cntList := make([]string, 0)
+
+	marker := azfile.Marker{}
+	for marker.NotDone() {
+		resp, err := fs.Service.ListSharesSegment(context.Background(), marker, azfile.ListSharesOptions{})
+		if err != nil {
+			log.Err("FileShare::ListContainers : Failed to get container list")
+			return cntList, err
+		}
+
+		for _, v := range resp.ShareItems {
+			cntList = append(cntList, v.Name)
+		}
+
+		marker = resp.NextMarker
+	}
+
+	return cntList, nil
 }
 
 // This is just for test, shall not be used otherwise
 func (fs *FileShare) SetPrefixPath(string) error {
-	return nil
+	return syscall.ENOTSUP
```

Comment:
Go ahead and implement this as part of your next PR - should be fairly simple, just setting a prefixPath variable